static/app/views/replays/detail/ai/useFetchReplaySummary.tsx

```diff
@@ -1,43 +1,178 @@
+  // data query was made before the start summary request.
+
+  // Otherwise, when start request query is finished, in the same render loop, we 1) invalidate the
+  // summary data query and 2) we have the stale version of the summary data. The consuming
+  // component will briefly show a completed state before the summary data query updates.
+  const startSummaryRequestTime = useRef<number>(0);
+
+  const segmentCount = replayRecord?.count_segments ?? 0;
+
+  const {
+    mutate: startSummaryRequestMutate,
+    isError: isStartSummaryRequestError,
+    isPending: isStartSummaryRequestPending,
+  } = useMutation({
+    mutationFn: () =>
+      api.requestPromise(
+        `/projects/${organization.slug}/${project?.slug}/replays/${replayRecord?.id}/summarize/`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          data: {
+            num_segments: segmentCount,
+          },
+        }
+      ),
+    onSuccess: () => {
+      // invalidate the query when a summary task is requested
+      // so the cached data is marked as stale.
+      queryClient.invalidateQueries({
+        queryKey: createAISummaryQueryKey(
+          organization.slug,
+          project?.slug,
+          replayRecord?.id ?? ''
+        ),
+      });
+      startSummaryRequestTime.current = Date.now();
+    },
+  });
+
+  const {
+    data: summaryData,
+    isPending,
+    isError,
+    dataUpdatedAt,
+  } = useApiQuery<SummaryResponse>(
     createAISummaryQueryKey(organization.slug, project?.slug, replayRecord?.id ?? ''),
     {
       staleTime: 0,
+      retry: false,
+      refetchInterval: query => {
+        if (isPolling(query.state.data?.[0], isStartSummaryRequestPending)) {
+          return POLL_INTERVAL_MS;
+        }
+        return false;
+      },
+      refetchOnWindowFocus: 'always',
       ...options,
     }
   );
+
+  const startSummaryRequest = useCallback(() => {
+    // Don't trigger if the feature is disabled
+    if (options?.enabled === false) {
+      return;
+    }
+
+    startSummaryRequestMutate();
+  }, [options?.enabled, startSummaryRequestMutate]);
+
+  return {
+    summaryData,
+    isPolling: isPolling(summaryData, isStartSummaryRequestPending),
+    isPending:
+      dataUpdatedAt < startSummaryRequestTime.current ||
+      isPending ||
+      summaryData?.status === ReplaySummaryStatus.PROCESSING ||
+      isStartSummaryRequestPending,
+    isError:
+      isError ||
+      summaryData?.status === ReplaySummaryStatus.ERROR ||
+      isStartSummaryRequestError,
+    startSummaryRequest,
+    isStartSummaryRequestPending,
+  };
 }
+
+const isPolling = (
+  summaryData: SummaryResponse | undefined,
+  isStartSummaryRequestPending: boolean
+) => {
+  if (!summaryData) {
+    // No data yet - poll if we've started a request
+    return isStartSummaryRequestPending;
```

Comment:
if there's no summary data and...

- start summary request is pending: it means we're kind of in a waiting state, so I think "polling = true" makes sense, we don't necessarily need to wait for start summary request to finish
- start summary request is *not* pending (meaning request is completed): we are in an incomplete state, summaryData needs to have a value

I think it makes sense to always return true in this case.